proposals/NNNN-lifetime-dependency.md

```diff
@@ -0,0 +1,501 @@
+# Compile-time Lifetime Dependency Annotations
+
+* Proposal: [SE-NNNN](NNNN-filename.md)
+* Authors: [Andrew Trick](https://github.com/atrick), [Meghana Gupta](https://github.com/meg-gupta), [Tim Kientzle](https://github.com/tbkka)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+* Review: ([pitch](https://forums.swift.org/t/pitch-non-escapable-types-and-lifetime-dependency/69865))
+
+## Introduction
+
+We would like to propose extensions to Swift's function-declaration syntax that allow authors to specify lifetime dependencies between the return value and one or more of the parameters.
+These would also be useable with methods that wish to declare a dependency on `self`.
+To reduce the burden of manually adding such annotations, we also propose inferring lifetime dependencies in certain common cases without requiring any additional annotations.
+
+This is a key requirement for the `StorageView` type (previously called `BufferView`) being discussed elsewhere, and is closely related to the proposal for `~Escapable` types.
```

Comment:
We have named this construct `Span`. Please let's avoid mentioning its deadnames; proposals are forever, and we really need to avoid indefinitely propagating these obsolete names.

```suggestion
This is a key requirement for the `Span` type that is being discussed elsewhere, and is closely related to the proposal for `~Escapable` types.
```